test_particle_in_field.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+from turbopy import Simulation
+from particle_in_field import EMWave, ChargedParticle, ForwardEuler
+
+owner_data = {"N": 10, "min": 20, "max": 30, "type": "test_pusher"}
+wave_data = {"amplitude": 10, "omega": 18}
+particle_data = {"position": 26.12, "pusher": "test_pusher"}
+
+
+@pytest.fixture(name="EM_example")
+def fixture_EM():
+    Owner = Simulation({"Grid": owner_data})
+    Owner.read_grid_from_input()
+    return EMWave(Owner, wave_data)
+
+
+def test_EMWave(EM_example):
+    assert isinstance(EM_example, EMWave)
+    assert EM_example.c == 2.998e8
+    assert EM_example.E0 == wave_data["amplitude"]
+    assert EM_example.omega == wave_data["omega"]
+    assert EM_example.k == wave_data["omega"] / 2.998e8
+
+
+@pytest.fixture(name="Charged_example")
+def fixture_Charged():
+    Owner = Simulation({"Grid": owner_data})
+    Owner.compute_tools.append(ForwardEuler(Owner, owner_data))
+    Owner.read_grid_from_input()
+    return ChargedParticle(Owner, particle_data)
```

Comment:
You won't need this if you make your fixture return the Simulation instance.

Side note about fixtures. They really are meant for taking care of "boilerplate" type stuff, so you don't have to repeat it over and over. If you find you are writing fixtures for each test, then you might as well save the trouble and just put the code in the test. What would be better, though, is to figure out what is common between the tests, and then put that _common_ code into a fixture.